settings/src/components/email-address.js

```diff
@@ -0,0 +1,94 @@
+
+/**
+ * WordPress dependencies
+ */
+import { Button, TextControl, Notice, Spinner } from '@wordpress/components';
+import { useState, useEffect } from '@wordpress/element';
+import { useSelect } from '@wordpress/data';
+import { store as coreDataStore, useEntityRecord } from '@wordpress/core-data';
+
+/**
+ * Render the Email setting.
+ */
+export default function EmailAddress( { userId } ) {
+	const { record, edit, save, editedRecord, hasEdits } = useEntityRecord( 'root', 'user', userId );
```

Comment:
Since this will be repeated on every screen, and it's already done in `Main`, maybe should just pass the entire result from `useEntityRecord` to `CurrentScreen` ?

Reply:
This seems like a reasonable suggestion, if the result set can be passed around like that.. in my early testing that wasn't possible, but it was more likely my misunderstanding of the state mechanisms.